src/components/templates/product-availability-modal/sales-channel-availability-table.tsx

```diff
@@ -0,0 +1,226 @@
+import { SalesChannel } from "@medusajs/medusa"
+import React, { useMemo } from "react"
+import { TableInstance } from "react-table"
+import Button from "../../fundamentals/button"
+import PlusIcon from "../../fundamentals/icons/plus-icon"
+import IndeterminateCheckbox from "../../molecules/indeterminate-checkbox"
+import { LayeredModalContext } from "../../molecules/modal/layered-modal"
+import Table, { TablePagination } from "../../molecules/table"
+import { useAddChannelsModalScreen } from "./add-channels-modal-screen"
+
+type SalesChannelAvailabilityTableProps = {
+  salesChannels: SalesChannel[]
+  limit: number
+  offset: number
+  setOffset: (offset: number) => void
+  setQuery: (query: string) => void
+  setSelectedRowIds?: (selectedRowIds: string[]) => void
+  tableAction?: React.ReactNode
+  tableState: TableInstance<SalesChannel>
+}
+
+type SalesChannelTableActionsProps = {
+  selectedRowIds: string[]
+  chosenSalesChannelIds: string[]
+  onAddSalesChannelsToSelected: (salesChannels: SalesChannel[]) => void
+  onDeselect: () => void
+  onRemove: () => void
+}
+
+export const useAvailableChannelsModalTableColumns = () => {
+  const columns = useMemo(
+    () => [
+      {
+        width: 30,
+        id: "selection",
+        Header: ({ getToggleAllPageRowsSelectedProps }) => (
+          <span className="flex justify-center w-[30px]">
+            <IndeterminateCheckbox {...getToggleAllPageRowsSelectedProps()} />
+          </span>
+        ),
+        Cell: ({ row }) => {
+          return (
+            <span
+              onClick={(e) => e.stopPropagation()}
+              className="flex justify-center w-[30px]"
+            >
+              <IndeterminateCheckbox {...row.getToggleRowSelectedProps()} />
+            </span>
+          )
+        },
+      },
+      {
+        Header: "Title",
+        accessor: "name",
+      },
+      {
+        Header: "Description",
+        accessor: "description",
+      },
+    ],
+    []
+  )
+
+  return [columns] as const
+}
+
+const SalesChannelAvailabilityTable: React.FC<SalesChannelAvailabilityTableProps> = ({
+  salesChannels,
+  limit,
+  offset,
+  setOffset,
+  setQuery,
+  tableState,
+  setSelectedRowIds,
+  tableAction,
+}) => {
+  const {
+    getTableProps,
+    getTableBodyProps,
+    headerGroups,
+    rows,
+    prepareRow,
+    canPreviousPage,
+    canNextPage,
+    pageCount,
+    nextPage,
+    previousPage,
+    // Get the state from the instance
+    state: { pageIndex, ...state },
+  } = tableState
+
+  const paginate = (direction: 1 | -1) => {
```

Comment:
suggestion: don't think we really need that and it could be moved closed to its usage wdyt? see comments below

Reply:
I have removed `paginate` and moved functionality into the `handleNext` and `handlePrev`